feature/gribi/ate_tests/hierarchical_weight_resolution_test/hierarchical_weight_resolution_test.go

```diff
@@ -270,40 +287,50 @@ func (a *attributes) configInterfaceDUT(t *testing.T, d *ondatra.DUTDevice, p *o
 		i = a.NewOCInterface(p.Name())
 	}
 
-	a.configSubinterfaceDUT(t, i)
-	intfPath := gnmi.OC().Interface(p.Name())
-	gnmi.Update(t, d, intfPath.Config(), i)
 	if *deviations.ExplicitPortSpeed {
-		fptest.SetPortSpeed(t, p)
+		i.GetOrCreateEthernet().PortSpeed = fptest.GetIfSpeed(t, p)
 	}
+
+	a.configSubinterfaceDUT(t, i)
+	intfPath := gnmi.OC().Interface(p.Name())
+	gnmi.Replace(t, d, intfPath.Config(), i)
 	fptest.LogQuery(t, "DUT", intfPath.Config(), gnmi.GetConfig(t, d, intfPath.Config()))
+
+	// assign subinterfaces to network-instances and update subif L3 configuration after that
+	a.assignSubifsToNetworkInstance(t, d, p)
+	a.configL3SubifDUT(t, i)
+	gnmi.Update(t, d, intfPath.Config(), i)
+
 }
 
 // configureNetworkInstance creates new Network Instance and configures it, if provided,
 // else configures the Default Network Instance.
-func (a *attributes) configureNetworkInstance(t *testing.T, d *ondatra.DUTDevice, p *ondatra.Port) {
+func (a *attributes) configureNetworkInstance(t *testing.T, d *ondatra.DUTDevice) {
 	t.Helper()
-	// Use default NI if not provided
-	if a.networkInstance != "" {
+	if a.networkInstance != *deviations.DefaultNetworkInstance {
 		ni := &oc.NetworkInstance{
 			Name: ygot.String(a.networkInstance),
 			Type: oc.NetworkInstanceTypes_NETWORK_INSTANCE_TYPE_L3VRF,
 		}
-		i := ni.GetOrCreateInterface(p.Name())
-		i.Interface = ygot.String(p.Name())
-		i.Subinterface = ygot.Uint32(0)
-
 		dni := gnmi.OC().NetworkInstance(a.networkInstance)
 		gnmi.Replace(t, d, dni.Config(), ni)
 		fptest.LogQuery(t, "NI", dni.Config(), gnmi.GetConfig(t, d, dni.Config()))
-	} else {
-		if *deviations.ExplicitInterfaceInDefaultVRF {
-			dni := gnmi.OC().NetworkInstance(*deviations.DefaultNetworkInstance)
-			gnmi.Replace(t, d, dni.Interface(p.Name()).Config(), &oc.NetworkInstance_Interface{
-				Id:           ygot.String(p.Name()),
-				Interface:    ygot.String(p.Name()),
-				Subinterface: ygot.Uint32(0),
-			})
+	}
+	if *deviations.ExplicitGRIBIUnderNetworkInstance {
+		fptest.EnableGRIBIUnderNetworkInstance(t, d, a.networkInstance)
+	}
+}
+
+func (a *attributes) assignSubifsToNetworkInstance(t *testing.T, d *ondatra.DUTDevice, p *ondatra.Port) {
+	niName := a.networkInstance
+
+	if niName != *deviations.DefaultNetworkInstance || *deviations.ExplicitInterfaceInDefaultVRF {
+		if a.numSubIntf == 0 {
+			fptest.AssignToNetworkInstance(t, d, p.Name(), niName, 0)
+		} else {
+			for i := uint32(1); i <= a.numSubIntf; i++ {
+				fptest.AssignToNetworkInstance(t, d, p.Name(), niName, i)
```

Comment:
This pr introduce a breakage for us. can u please add a deviation in AssignToNetworkInstance to use the interface id exactly as interface name for subinterface zero. 

```
	if si==0 & *deviations.SubInterfaceIDZero{
		netInstIntf.Id = ygot.String(intf.GetName()))
	} else {
		  netInstIntf.Id = ygot.String(intf.GetName() + "." + fmt.Sprint(si))
	}